kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,284 @@
+//! For now we just use simple functions to deserialize table properties from strings. This allows
+//! us to relatively simply implement the functionality described in the protocol and expose
+//! 'simple' types to the user in the [`TableProperties`] struct. E.g. we can expose a `bool`
+//! directly instead of a `BoolConfig` type that we implement `Deserialize` for.
+
+use crate::expressions::ColumnName;
+
+use std::time::Duration;
+
+use serde::de::{self, Deserializer};
+use serde::Deserialize;
+
+const SECONDS_PER_MINUTE: u64 = 60;
+const SECONDS_PER_HOUR: u64 = 60 * SECONDS_PER_MINUTE;
+const SECONDS_PER_DAY: u64 = 24 * SECONDS_PER_HOUR;
+const SECONDS_PER_WEEK: u64 = 7 * SECONDS_PER_DAY;
+
+/// Transparently pass through the option deserialization to the inner type.
+pub(crate) fn deserialize_option<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    T::deserialize(deserializer).map(Some)
```

Comment:
What code path produces `None`? This one only seems to produce `Some`?

Reply:
welp this was the straw that broke `serde`'s back.. instead of implementing a custom `Deserialize` I just decided to do the naive thing and implement a big match by hand. Actually isn't _that_ gross and we can always try `serde` again or just write our own macro to make everything nice